app/helpers/tags.rb

```diff
@@ -0,0 +1,13 @@
+# needs to check for both duplicate tags AND duplicate tagging objects associated
+# with question.
+def add_tags (question, tag_string)
+  tag_string.split(" ").each do |tag_word|
+    tag = Tag.find_or_create_by(word: tag_word)
+
+    #check for tagging on this object.
+    if !Tagging.find_by(question: question, tag: tag)
+      #if question has not already been tagged by this object, then tag
+      question.tags << tag
+    end
+  end
+end
```

Comment:
So this method takes a question and a tag string and adds tags to the question. This definitely feels like a method that belongs on the question itself, as `question.add_tags(tag_string)` is a more natural way to handle this functionality. Plus, helpers are usually just things you use in your view.